hsc/crawler.py

```diff
@@ -10,23 +10,62 @@ class Crawler():
 	domain_url = base_url + 'domains/{}/{}'
 	problem_url = base_url + 'challenges/{}/problem'
 
-	new_readme_text = '## [{}]({})\n\nProblem Name|Problem Link|Solution Link\n---|---|---'
-	problem_readme_text = '{}|[Problem]({})|[Solution](./{}{})'
+	new_readme_text = '## [{}]({})\n\n|Problem Name|Problem Link|Solution Link|\n|---|---|---|\n'
+	problem_readme_text = '|{}|[Problem]({})|[Solution](./{})|\n'
 
 	base_folder_name = 'Hackerrank'
-
-	# add other exclusive extensions if your data not crawled properly
-	special_extensions = {
+	make_language_folder = False
+
+	# file extensions
+	file_extensions = {
+		'ada': 'ada',
+		'bash': 'sh',
+		'c': 'c',
+		'clojure': 'clj',
+		'coffeescript': 'coffee',
+		'cpp': 'cpp',
 		'cpp14': 'cpp',
+		'csharp': 'cs',
+		'd': 'd',
+		'db2': 'sql',
+		'elixir': 'ex',
+		'erlang': 'erl',
+		'fortran': 'for',
+		'fsharp': 'fs',
+		'go': 'go',
+		'groovy': 'groovy',
 		'haskell': 'hs',
+		'java': 'java',
 		'java8': 'java',
+		'javascript': 'js',
+		'julia': 'jl',
+		'kotlin': 'kt',
+		'lolcode': 'lol',
+		'lua': 'lua',
 		'mysql': 'sql',
+		'objectivec': 'm',
+		'ocaml': 'ml',
+		'octave': 'oct',
 		'oracle': 'sql',
+		'pascal': 'pas',
 		'perl': 'pl',
+		'php': 'php',
+		'pypy': 'py',
+		'pypy3': 'py',
 		'python': 'py',
 		'python3': 'py',
+		'racket': 'rkt',
+		'r': 'r',
+		'ruby': 'rb',
 		'rust': 'rs',
-		'text': 'txt',
```

Comment:
Solution file can be in text format also. I think we can keep this.

Reply:
How can it be a text file? Even if it is, its default extension will be .text which should be fine.

Reply:
I haven't seen the extension of a text file as `.text`

Reply:
Marking this as resolved, as we weren't able to find any such solution which might have a text file as the extension.